sov-modules/sov-state/src/map.rs

```diff
@@ -43,18 +43,23 @@ impl<K: Encode, V: Encode + Decode, S: Storage> StateMap<K, V, S> {
         })
     }
 
-    // Removes a key from the StateMap, returning the corresponding value (or None if the key is absent).
+    /// Removes a key from the StateMap, returning the corresponding value (or None if the key is absent).
     pub fn remove(&mut self, key: &K, working_set: &mut WorkingSet<S>) -> Option<V> {
         working_set.remove_value(self.prefix(), key)
     }
 
-    // Removes a key from the StateMap, returning the corresponding value (or Error if the key is absent).
+    /// Removes a key from the StateMap, returning the corresponding value (or Error if the key is absent).
     pub fn remove_or_err(&mut self, key: &K, working_set: &mut WorkingSet<S>) -> Result<V, Error> {
         self.remove(key, working_set).ok_or_else(|| {
             Error::MissingValue(self.prefix().clone(), StorageKey::new(self.prefix(), key))
         })
     }
 
+    /// Deletes a key from the StateMap.
+    pub fn delete(&mut self, key: &K, working_set: &mut WorkingSet<S>) {
```

Comment:
Can you please add a test for it in `state_tests.rs`
For example you can create a new test based on:
```
#[test]
fn test_state_value() {
    let path = schemadb::temppath::TempPath::new();
    for (before_remove, after_remove) in create_storage_operations() {
        let value = 11;
        let (mut state_value, mut working_set) = create_state_value_and_storage(value, &path);

        working_set = before_remove.execute(working_set);
        assert_eq!(state_value.remove(&mut working_set).unwrap(), value);

        working_set = after_remove.execute(working_set);
        assert!(state_value.get(&mut working_set).is_none())
    }
}```

and change `remove` to `delete.`

Reply:
Thx!, LGTM. I will merge after the CI finishes.